src/components/Button/index.tsx

```diff
@@ -1,36 +1,45 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import classNames from 'classnames'
-import { Tooltip } from './Tooltip'
-import './styles/button.less'
+
+import { Tooltip } from '../Tooltip'
+
+import underscore from '../../utils/underscore'
+
+import styles from './Button.module.less'
 
 type NativeButtonProps = React.DetailedHTMLProps<
     React.ButtonHTMLAttributes<HTMLButtonElement>,
     HTMLButtonElement
 >
 
-export type ButtonProps = Omit<NativeButtonProps, 'title' | 'ref'> & {
-    /**
-     * Loading style. When true it disables the button, but it also adds a visual indication of
-     * some in-progress operation going on.
-     */
-    loading?: boolean
-    /**
-     * Controls visually how the button shows up from a predefined set of kinds of buttons.
-     */
-    variant?: 'primary' | 'secondary' | 'danger' | 'link'
-    /**
-     * The size of the button. If not provided it shows up in a normal size.
-     */
-    size?: 'default' | 'small' | 'large'
-    /**
-     * Tooltip that is displayed on hover.
-     *
-     * This replaces `title` which is not supported for these buttons to avoid confusion.
-     */
-    tooltip?: React.ReactNode
+type ThemeProps = {
+    theme?: 'twistLight' | 'twistDark'
 }
 
+export type ButtonProps = Omit<NativeButtonProps, 'title' | 'ref'> &
+    ThemeProps & {
```

Comment:
I raised this before and I'll raise it again. I am not a fan of us being able to override the theme on a per component basis.

I would expect the components to read the theme inside them.

Reply:
I've found that it is useful to override themes in certain scenarios. One example that comes to mind is in case of a lightbox component that behaves as if it's in a dark theme even inside light-theme contexts. Otherwise (as in the case of twist right now) we'll need to add lightbox-specific variables to the theme. 

I borrowed this pattern from [styled-components](https://styled-components.com/docs/advanced#the-theme-prop). This provides us additional flexibility, and doesn't take anything away. ThemeProvider is still available, but because of how it works, it cannot be nested within each other.

Reply:
After discussion with Steven and Henning, we might need to switch to a CSS class-based theme switcher, and where the styles LESS file has duplicate definitions for each CSS theme-class. This means that we'll have extra "un-needed" (for themes that are not the current theme) CSS loaded on every page. But I suppose that's the only way to make this work.

cc @gnapse @henningmu @stkao05